### PR TITLE
pkgsStatic.perl: fix build

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

I don't understand how this worked prior to 1dbb306 ("pkgsStatic.perl: Use \`dontAddStaticConfigureFlags\` instead of overriding"), but what I do know is that it's very broken now, to the point that perl-cross needs a patch.

I doubt the -Duseshrplib is actually necessary when native compiling, but removing it would be a mass rebuild, so I've left it in place for now.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-burns** [comment]
pkgsCross.aarch64-multiplatform.perl failed:
```
Running Mkbootstrap for B ()
Can't locate DynaLoader.pm in @INC (you may need to install the DynaLoader module) (@INC contains: /build/perl-5.34.0/cnf/stub /build/perl-5.34.0/cnf/cpan /build/perl-5.34.0/cpan/AutoLoader/lib /build/perl-5.34.0/dist/Exporter/lib /build/perl-5.34.0/dist/Cwd /build/perl-5.34.0/dist/Cwd/lib /build/perl-5.34.0/cpan/ExtUtils-Command/lib /build/perl-5.34.0/cpan/ExtUtils-Install/lib /build/perl-5.34.0/cpan/ExtUtils-MakeMaker/lib /build/perl-5.34.0/dist/ExtUtils-MakeMaker/lib /build/perl-5.34.0/cpan/ExtUtils-Manifest/lib /build/perl-5.34.0/cpan/File-Path/lib /build/perl-5.34.0/cpan/Pod-Simple/lib /build/perl-5.34.0/cpan/Pod-Escapes/lib /build/perl-5.34.0/ext/File-Find/lib /build/perl-5.34.0/ext/re /build/perl-5.34.0/cpan/ExtUtils-Constant/lib /build/perl-5.34.0/dist/ExtUtils-ParseXS/lib /build/perl-5.34.0/dist/constant/lib /build/perl-5.34.0/cpan/Getopt-Long/lib /build/perl-5.34.0/cpan/Text-Tabs/lib /build/perl-5.34.0/dist/Carp/lib /build/perl-5.34.0/cpan/podlators/lib /build/perl-5.34.0/cpan/parent/lib /build/perl-5.34.0/cpan/version/lib /build/perl-5.34.0/dist/Pod-Simple/lib /build/perl-5.34.0/dist/if /build/perl-5.34.0/lib ../../lib .) at /build/perl-5.34.0/cpan/ExtUtils-MakeMaker/lib/ExtUtils/Mkbootstrap.pm line 28.
make[2]: *** [Makefile:451: B.bs] Error 2
make[2]: Leaving directory '/build/perl-5.34.0/ext/B'
make[1]: *** [Makefile:235: ext/B/pm_to_blib] Error 2
make[1]: Leaving directory '/build/perl-5.34.0'
make: *** [Makefile:81: all] Error 2
builder for '/nix/store/vl4fy433w7fmnk1vdsfw2r3933dxpxda-perl-5.34.0-aarch64-unknown-linux-gnu.drv' failed with exit code 2
error: build of '/nix/store/vl4fy433w7fmnk1vdsfw2r3933dxpxda-perl-5.34.0-aarch64-unknown-linux-gnu.drv' failed
```

**Comment by alyssais** [comment]
@OfBorg build pkgsStatic.perl pkgsStatic.aarch64-multiplatform.perl

**Review by stigtsp — APPROVED** [review]
LGTM
- Basic test of `pkgsStatic.perl` and `pkgsCross.aarch64-multiplatform.perl` OK

File: pkgs/development/interpreters/perl/default.nix [hunk 1]
```diff
@@ -71,13 +78,14 @@ let
       ++ [
         "-Uinstallusrbinperl"
         "-Dinstallstyle=lib/perl5"
-        "-Duseshrplib"
+      ] ++ lib.optional (!crossCompiling) "-Duseshrplib" ++ [
```
Comment (Artturin) [inline pkgs/development/interpreters/perl/default.nix:81]: why not `!stdenv.hostPlatform.isStatic`?

**Pull request merged**